apiserver/facades/controller/crossmodelrelations/crossmodelrelations.go

```diff
@@ -176,12 +177,19 @@ func (api *CrossModelRelationsAPI) registerRemoteRelation(relation params.Regist
 		return nil, errors.Annotate(err, "loading model")
 	}
 	offerURL := crossmodel.MakeURL(model.Owner().Name(), model.Name(), relation.OfferName, "")
-	if err := api.checkMacaroons(relation.Macaroons, map[string]string{
+	attr, err := api.checkMacaroons(relation.Macaroons, map[string]string{
 		"source-model-uuid": api.st.ModelUUID(),
 		"offer-url":         offerURL,
-	}); err != nil {
+	})
+	if err != nil {
 		return nil, err
 	}
+	// The macaroon needs to be attenuated to a user.
+	username, ok := attr["username"]
```

Comment:
You can't do this in the required values because that can only check for a specific username, where you want just _some_ username, is that right? And because the macaroon is made by the offering controller, there's no link between the user's name here and the user's name in the macaroon? (Just checking that I understand what's going on.)